pkg/registry/pod/etcd/etcd_test.go

```diff
@@ -255,10 +245,10 @@ func TestResourceLocation(t *testing.T) {
 
 	ctx := api.NewDefaultContext()
 	for _, tc := range testCases {
-		storage, _, _, fakeClient := newStorage(t)
+		storage, _, _, server := newStorage(t)
```

Comment:
Can you please opaque the whole body of the loop into 
func() {
}()
and the use defer server.Terminate() just after this line?

I think this is more robust.